### DISCUSSION TITLE
View weather for a location without adding it

### DISCUSSION THREAD

**Discussion by Neluji** [breezy-weather/breezy-weather#1504] [Ideas]
### Describe your idea

The ability to view the weather for a location without adding it to the list of saved locations (i.e. a one-time view) could be useful to look it up for a location that we will maybe only need once (e.g. when traveling).  
We would search for the location, select it and it will display the weather using the default setting (or a separate option) for the sources. Maybe a button could be present to add it to the list of saved locations.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
There are a few reasons it's not done that way.

Not saving the location means you're by-passing the [caching rules](https://github.com/breezy-weather/breezy-weather/blob/main/docs/UPDATES.md#caching). For example, if you're checking the weather for a city just one time, and then 30 sec later "What was the precipitation forecast already?" then you check again the city and make all the requests all over again instead of using cached data.

Then, there are the use cases. If you are traveling to a city just for 1 day or a few days, you're not just checking it once. There's a high chance you will check it a few times for all the period you will spend there. And it's just more convenient to add the city, and just tap on it whenever you need it, or reorder it first for the time you spend there, instead of searching for it every time you need it. Then, whenever you don't need this city anymore, it's just a swipe away to delete it. FYI, [secondary locations are usually not updated in the background](https://github.com/breezy-weather/breezy-weather/blob/main/docs/UPDATES.md#background-or-automatic-weather-update), so they should not make additional useless calls if you save the temporary city.

Finally, there is another technical reason. The app is designed to save the data in database for offline use, and changing the logic to allow for online-only data would require significant changes for something rather niche, that can be easily workarounded (= if you absolutely need a city only once, you can save it and delete it immediately. Unless I'm missing something, it would be just 2-3 additional seconds once in a while. If it's something happening every day, then you should save the city for re-use).
